wordpress-mobile/WordPress-iOS
# Fixes a problem with posts losing their local changes.

Fixes https://github.com/wordpress-mobile/WordPress-iOS/issues/4904 and https://github.com/wordpress-mobile/WordPress-iOS/issues/404 (only the first part of it).

Also makes local revisions much more resilient to these kind of issues in the future.  Local changes that are not in the server will also be tagged now as "Local" regardless of if the post was previously published or not.

**Test 1:**
1. Launch the app.
2. Create a new post.
3. Publish it.
4. Edit the post.
5. Disconnect from the internet.
6. Update the post, you should see a failure message.

Expected behavior: the post should be marked as local in the list of posts.  The original should not be visible.

**Test 2:**
1. Follow all the steps from test 1.
2. Once you see the post marked as "Local", pull down the list to refresh.  The post should remain unaffected by the action.
3. Edit the original post using your web browser and save it.
4. Pull down to refresh the list of posts again.

Expected behaviour: the local post should not be affected by either the pull to refresh action, or by changes applied through the web interface.

**Test 3:**
Make sure that posts tagged as local always tell you there are unsaved changes when you open them.

**Known issues:**
There are some refreshing issues in the list of posts that will be handled in a different PR.  The refreshing issues existed before this PR but may seem to come up more often due to the change in GUI logic.

Needs review: @aerych
